lib/blocs/authentication/viewmodels/authentication_bloc.dart

```diff
@@ -24,6 +24,7 @@ class AuthenticationBloc extends Bloc<AuthenticationEvent, AuthenticationState>
     }
     if (event is OnImportAccount) {
       settingsStorage.saveAccount(event.account, event.privateKey);
+      settingsStorage.inRecoveryMode = false; // clear recovery flag - new account was imported.
```

Comment:
how about we move all the changes done when the account is imported into a method in the `settingsStorage`

settingsStorage.saveAccount() could also do this in the implementation 
settingsStorage.inRecoveryMode = false
 settingsStorage.privateKeyBackedUp = true;

Reply:
I was looking into that but saveAccount is called from a few different places & looked like that would be incorrect for some cases.

Need to map out the sequence of events in recovery...

Here's a list of saveAccount cases - we could just make different methods in settingsStorage that sets the correct flags

1. saveAccount from creating an account via invite -> private key not backed up, recovery -> false
2. Recovery saveAccount: Creating a new account and waiting for recovery to finish -> private key not backed up, recovery -> true - this is when we create a new keypair and wait for guardians to sign
3. saveAccount from importing a key -> private key backed up -> true (since the key was imported!), recovery -> false